src/input/keyboard/mod.rs

```diff
@@ -249,12 +251,166 @@ impl<'a> KeysymHandle<'a> {
             .key_get_syms_by_level(self.keycode, self.state.key_get_layout(self.keycode), 0)
     }
 
+    /// Get the raw latin keysym or fallback to current raw keysym.
+    ///
+    /// This method is handy to implement layout agnostic bindings. Keep in mind that
+    /// it could be not-ideal to use just this function, since some layouts utilize non-standard
+    /// shift levels and you should look into [`Self::modified_sym`] first.
+    ///
+    /// The `None` is returned when the underlying keycode doesn't produce a valid keysym.
+    pub fn raw_latin_sym_or_raw_current_sym(&self) -> Option<Keysym> {
+        let effective_layout = Layout(self.state.key_get_layout(self.keycode));
+        // NOTE: There's always a keysym in the current layout given that we have modified_sym.
+        let base_sym = *self.raw_syms().first()?;
+
+        // If the character is ascii or non-printable, return it.
+        if base_sym.key_char().map(|ch| ch.is_ascii()).unwrap_or(true) {
+            return Some(base_sym);
+        };
+
+        // Try to look other layouts and find the one with ascii character.
+        for layout in self.layouts() {
+            if layout == effective_layout {
+                continue;
+            }
+
+            for keysym in self.raw_syms_for_key_in_layout(self.keycode, layout) {
+                // NOTE: Only check for ascii alphabetic since the rest of the punctuation should
+                // be uniform across layouts.
+                if keysym
+                    .key_char()
+                    .map(|key| key.is_ascii_alphabetic())
+                    .unwrap_or(false)
+                {
+                    return Some(*keysym);
+                }
+            }
+        }
+
+        Some(base_sym)
```

Comment:
I decided to go with something like that for now, it sort of works, even though not ideal, but it should be enough to not trigger bindings you don't want to trigger, like due to different `Enter` locations. Not sure how it's relevant though.

There's no helper to check for latin, but `key_char` logic sort of does that already.